### PR TITLE
Add a try_from_js implementation for Vec<T> (accept any Array-like)

### PR DESCRIPTION
**Pull request by hansl**
This PR adds an implementation of `TryFromJs` for `Vec<T: TryFromJs>` to create Rust vectors from Array-like JavaScript objects.

This is useful when using typed arguments to native functions with arrays.

### CODE REVIEW

File: core/engine/src/value/conversions/try_from_js.rs
Comment (jedel1043) [inline core/engine/src/value/conversions/try_from_js.rs:79]: I'd avoid casting to `usize` and use `usize::try_from` with some added context in case the conversion fails.

**Review by jedel1043 — CHANGES_REQUESTED** [review]
Thank you! I just have a small suggestion, but everything else looks good!

**Comment by Razican** [comment]
The only concern here would be on how to implement this for TypedArrays, which should have a much faster conversion. Not sure if we would need to specialize or we could just add another branch.

**Comment by hansl** [comment]
> The only concern here would be on how to implement this for TypedArrays

I suggest a specialized type, like `impl TryFromJs for Bytes`. There's no easy way to do that in Rust yet (see how serde handles `Vec<u8>` for example). Maybe when we have trait specialization in stable.

File: core/engine/src/value/conversions/try_from_js.rs
Comment (hansl) [reply core/engine/src/value/conversions/try_from_js.rs]: Done.

**Review by nekevss — APPROVED** [review]
Looks good to me! 😄 Had one nit, but that's not blocking merging.

File: core/engine/src/value/conversions/try_from_js.rs
Comment (hansl) [reply core/engine/src/value/conversions/try_from_js.rs:288]: Done. What could I do to unblock this? It's blocking some of my follow work that I'd like to submit.

**Review by jedel1043 — APPROVED** [review]
Looks good! Thank you for the contribution!

**Pull request merged**
